File: Models/Supplier.cs
using System.ComponentModel.DataAnnotations;

namespace EventPlanner.Models
{
    public class Supplier
    {
        [Key]
        public Guid Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public SupplierType Type { get; set; }
        public string ContactPerson { get; set; }
        [DataType(DataType.PhoneNumber)]
        public string Mobile { get; set; }
        [DataType(DataType.EmailAddress)]
        public string Email { get; set; }
        public IList<Attachment> Attachments { get; set; }
        public bool DownPaymentRequired { get; set; }
        [DataType(DataType.Currency)]
        public decimal PackagePrice { get; set; }
        [DataType(DataType.Currency)]
        public decimal Discount { get; set; }
        [DataType(DataType.Currency)]
        public decimal OtherPayments { get; set; }
        [DataType(DataType.Currency)]
        public decimal FirstDownPayment { get; set; }
        [DataType(DataType.Currency)]
        public decimal SecondDownPayment { get; set; }
        [DataType(DataType.Currency)]
        public decimal ThirdDownPayment { get; set; }
        [DataType(DataType.Currency)]
        public decimal TotalDown { get; set; }
        [DataType(DataType.Currency)]
        public decimal Balance { get; set; }
        public string Remarks { get; set; }
        public Event Event { get; set; }
    }
}

File: ClientApp/src/app/supplier/supplier.ts
import { SupplierType } from '../supplier-types/supplier-type.model';
import { Event } from '../event/event';
import { Attachment } from '../supplier-form/attachment';

export class Supplier {
    id: string;
    name: string;
    type: SupplierType;
    contactPerson: string;
    mobile: string;
    email: string;
    attachments: Attachment[];
    downPaymentRequired: boolean;
    packagePrice: number;
    discount: number;
    otherPayments: number;
    firstDownPayment: number;
    secondDownPayment: number;
    thirdDownPayment: number;
    totalDown: number;
    balance: number;
    remarks: string;
    event: Event;

    constructor() { }
}
